core/kmod/sn_netdev.c

```diff
@@ -913,10 +905,16 @@ void sn_release_netdev(struct sn_device *dev)
 	rtnl_lock();
 
 	/* it is possible that the netdev has already been unregistered */
-	if (dev && dev->netdev && dev->netdev->reg_state == NETREG_REGISTERED)
```

Comment:
Is it safe to assume `dev` is always non-null by the time we get here? Would it be worth checking and returning early before taking the lock?

Reply:
I believe it's safe... Not sure why I checked it in the past. As per your suggestion I added an early check, it should be better safe than sorry.